superset/dashboards/api.py

```diff
@@ -1364,7 +1364,7 @@ def delete_embedded(self, dashboard: Dashboard) -> Response:
               $ref: '#/components/responses/500'
         """
         for embedded in dashboard.embedded:
-            DashboardDAO.delete(embedded)
+            EmbeddedDashboardDAO.delete(embedded)
```

Comment:
I'm somewhat surprised that https://github.com/apache/superset/pull/24465 didn't pick this up.